### ISSUE TITLE
[java] Incorrect generics for primitives

### ISSUE THREAD

**Issue by adamretter** [HaxeFoundation/haxe#4203]
The following trivial code leads to uncompilable Java code:

```haxe
@:nativeGen
interface NativeItem<T> {
    public function getIt() : T;
}

class Thing1 implements NativeItem<Bool> {
    var value: Bool;
    public function new(value) {
        this.value = value;
    }

    public function getIt() : Bool {
        return value;
    }
}
```

The relevant resultant Java source code which cannot be compiled:

```java
public class Thing1 extends haxe.lang.HxObject implements xpdl.xdm.NativeItem<java.lang.Object> {

   public boolean value;  

    public boolean getIt() {
                //line 30 ...
                return this.value;
    }

}
```

I guess Haxe does not know how to turn its `Bool` type into a Java object type (as it already maps to a java primitive type, which can't be used in generics), and so defaults to `java.lang.Object`. It should not do this... instead for primitive types it should default to the equivalent object type, e.g. `java.lang.Boolean` when outputting Java generics. javac will then magically handle the boxing/unboxing between primitive and object types.

Also I might add that there is no `@Override` annotation on the generated `getIt` function, I think there should be.

In summary, the **correct** resultant Java code should instead look like -

```java
public class Thing1 extends haxe.lang.HxObject implements xpdl.xdm.NativeItem<java.lang.Boolean> {

   public boolean value;  

    @Override
    public java.lang.Boolean getIt() {
                //line 30 ...
                return this.value;
    }

}
```

**Comment by waneck** [comment]
Seems to be fixed already

**Comment by Simn** [comment]
The test you added actually fails:

```
src/unit/issues/_Issue4203/Thing1.java:7: error: Thing1 is not abstract and does not override abstract method getIt() in NativeItem
public class Thing1 extends haxe.lang.HxObject implements unit.issues._Issue4203.NativeItem<java.lang.Object>
       ^
src/unit/issues/_Issue4203/Thing1.java:44: error: getIt() in Thing1 cannot implement getIt() in NativeItem
	public boolean getIt()
	               ^
  return type boolean is not compatible with Object
  where T is a type-variable:
    T extends Object declared in interface NativeItem
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
2 errors
Compilation error
Native compilation failed
```

**Comment by waneck** [comment]
Yeah I forgot to test with -dce no! OOPS
